sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java

```diff
@@ -0,0 +1,482 @@
+ *
+ * @see ContainerRegistryBlobClientBuilder
+ */
+@ServiceClient(builder = ContainerRegistryBlobClientBuilder.class, isAsync = true)
+public class ContainerRegistryBlobAsyncClient {
+
+    private final AzureContainerRegistryImpl registryImplClient;
+    private final ContainerRegistryBlobsImpl blobsImpl;
+    private final ContainerRegistriesImpl registriesImpl;
+    private final HttpPipeline httpPipeline;
+    private final String endpoint;
+    private final String apiVersion;
+    private final String repositoryName;
+    private final String registryLoginServer;
+
+    private final ClientLogger logger = new ClientLogger(ContainerRegistryBlobAsyncClient.class);
+
+    ContainerRegistryBlobAsyncClient(String repositoryName, HttpPipeline httpPipeline, String endpoint, String version) {
+        this.repositoryName = repositoryName;
+        this.httpPipeline = httpPipeline;
+        this.endpoint = endpoint;
+        this.registryImplClient = new AzureContainerRegistryImplBuilder()
+            .url(endpoint)
+            .pipeline(httpPipeline)
+            //.apiVersion(version)
+            .buildClient();
+        this.blobsImpl = this.registryImplClient.getContainerRegistryBlobs();
+        this.registriesImpl = this.registryImplClient.getContainerRegistries();
+        this.apiVersion = version;
+        try {
+            URL endpointUrl = new URL(endpoint);
+            this.registryLoginServer = endpointUrl.getHost();
+        } catch (MalformedURLException ex) {
+            // This will not happen.
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL"));
+        }
+    }
+
+    /**
+     * This method returns the registry's repository on which operations are being performed.
+     *
+     * @return The name of the repository
+     */
+    public String getRepositoryName() {
+        return this.repositoryName;
+    }
+
+    /**
+     * This method returns the complete registry endpoint.
+     *
+     * @return The registry endpoint including the authority.
+     */
+    public String getEndpoint() {
+        return this.endpoint;
+    }
+
+    /**
+     * Upload the Oci manifest to the repository.
+     * The upload is done as a single operation.
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     * @param manifest The OciManifest that needs to be uploaded.
+     * @return operation result.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code manifest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<UploadManifestResult> uploadManifest(OciManifest manifest) {
+        if (manifest == null) {
+            throw logger.logExceptionAsError(new NullPointerException("'manifest' can't be null."));
```

Comment:
This should be returned as a `Mono.error`, same for other locations in this file that throw